packages/create-plugin/src/migrations/utils.ts

```diff
@@ -132,3 +133,174 @@ export function installNPMDependencies(context: Context) {
     execSync(installCmd, { cwd: context.basePath, stdio: 'inherit' });
   }
 }
+
+export function readJsonFile<T extends object = any>(context: Context, path: string): T {
+  if (!context.doesFileExist(path)) {
+    throw new Error(`Cannot find ${path}`);
+  }
+  try {
+    return JSON.parse(context.getFile(path) || '{}');
+  } catch (e) {
+    throw new Error(`Cannot parse ${path}: ${e}`);
+  }
+}
+
+export function addDependenciesToPackageJson(
+  context: Context,
+  dependencies: Record<string, string>,
+  devDependencies: Record<string, string> = {},
+  packageJsonPath = 'package.json'
+) {
```

Comment:
What are the main arguments for using homegrown solution that we need to maintain instead of using `npm`|`yarn`|`pnpm` or 3rd party lib?

Reply:
Thanks for the great explain, all valid and solid points 🙌